### PR TITLE
feat:[CDS-84539]: Generic Harness Store Download Step NG Manager Change

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

This is a generic harness store download step which will be used to harness store contents inside a container step group. For more details , please refer this doc

https://harness.atlassian.net/wiki/spaces/DEV/pages/21551744043/Download+Manifest+From+Harness+Store+in+CD+Container+Step+Group

Cases Tested:

- Downloading a single file from project scope downloads the file in downloadPath
- Downloading a single file from org scope downloads the file in downloadPath
- Downloading a single file from account scope downloads the file in downloadPath
- Downloading a folder from project scope downloads it recursively with all sub folders created in downloadPath
- Downloading a folder from org scope downloads it recursively with all sub folders created in downloadPath
- Downloading a folder from account scope downloads it recursively with all sub folders created in downloadPath
- Downloading it in /harness/<StepIdentifier> when downloadPath not present

https://github.com/harness/harness-core/assets/100153994/108aacd2-a7d9-4404-8806-e3e8483f8053

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by ivan-83 — CHANGES_REQUESTED** [review]
> Reviewed 24 of 33 files at r1, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)* status: 33 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/config.yml` line 556 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFksuy4kuxxbLsI3Wx:-NlFksuz5b-ZY0FneClL:b-130udn) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/120-ng-manager/config.yml#L556)):*
> ```YAML
>   downloadHarnessStoreConfig:
>     # Use rootless image only for K8 download aws s3
>     image: harnessdev/download-harness-store:1.0.0-rootless-linux
> ```

Minor: download-harness-store >> download-harness-file-store or download-harness-fs
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFnO6o3hWMEFwN-whZ:-NlFnO6o3hWMEFwN-wh_:bgqlfxc) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L38)):*
> ```Java
>   @Inject private Supplier<DelegateCallbackToken> delegateCallbackTokenSupplier;
> 
>   @Inject private DownloadHarnessStoreStepHelper downloadHarnessStoreStepHelper;
> ```

Minor: DownloadHarnessStoreStepHelper >> DownloadHarnessFileStoreStepHelper
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 45 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFnboRA1Ids2yn9rGt:-NlFnboRA1Ids2yn9rGu:bz56cod) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L45)):*
> ```Java
> 
>   public static final StepType STEP_TYPE = StepType.newBuilder()
>                                                .setType(ExecutionNodeType.DOWNLOAD_HARNESS_STORE.getYamlType())
> ```

I think it's important to emphasize what type of store (blob, file, queue) in our case it's a file. Pls, rename all paces accordingly. 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 62 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFmTJc0bLZFc4EHSMj:-NlFmTJc0bLZFc4EHSMk:bte3dcc) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L62)):*
> ```Java
>   public UnitStep getSerialisedStep(Ambiance ambiance, StepElementParameters stepElementParameters, String accountId,
>       String logKey, long timeout, String parkedTaskId) {
>     DownloadHarnessStoreStepParameters downloadHarnessStoreStepParameters =
> ```

Minor: DownloadHarnessStoreStepParameters >> DownloadHarnessFileStoreStepParameters, 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java` line 45 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFpXkM9yFiy4jQvxe9:-NlFpXkM9yFiy4jQvxeA:b-hdctyv) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java#L45)):*
> ```Java
> 
>     HashMap<String, String> envVarsMap = new HashMap<>();
>     envVarsMap.put("PLUGIN_HARNESS_STORE_FILE_MAP", filesJson);
> ```

PLUGIN_HARNESS_STORE_FILE_MAP >> PLUGIN_HARNESS_FILE_STORE_MAP
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java` line 63 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFrher-yHXaVyfjuw-:-NlFrher-yHXaVyfjuw0:b-2fbnbv) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java#L63)):*
> ```Java
>   }
> 
>   public Map<String, String> getFileContentsFromManifest(NGAccess ngAccess, List<String> scopedFilePathList) {
> ```

Move this method to the utility class, I guess we already have this method, try to reuse it.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java` line 94 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFrwqk83SczVBdF9ru:-NlFrwqk83SczVBdF9rv:b-yojev8) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStepHelper.java#L94)):*
> ```Java
>   }
> 
>   public String convertToJson(Map<String, String> files) {
> ```

The same as for the above method.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1626 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFt5Rs2DpmGH05ZjfE:-NlFt5Rs2DpmGH05ZjfF:bmbq04b) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1626)):*
> ```Java
> 
>     StepInfo downloadHarnessStore = StepInfo.newBuilder()
>                                         .setName("DOWNLOAD Harness Store")
> ```

DOWNLOAD Harness Store >> Download Harness File Store"
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/containerStepGroup/DownloadHarnessStoreStepVariableCreator.java` line 23 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFu62s0Ge_mlBJYlXF:-NlFu62s0Ge_mlBJYlXG:b4n33fm) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/containerStepGroup/DownloadHarnessStoreStepVariableCreator.java#L23)):*
> ```Java
>   @Override
>   public Set<String> getSupportedStepTypes() {
>     return Collections.singleton(StepSpecTypeConstants.DOWNLOAD_HARNESS_STORE);
> ```

The same comment for renaming DOWNLOAD_HARNESS_STORE >> DOWNLOAD_HARNESS_FILE_STORE
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java` line 77 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFv6Qr6CYMVQRYpiA3:-NlFv6Qr6CYMVQRYpiA4:b6i2nni) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java#L77)):*
> ```Java
>     pluginDetailsBuilder.putAllEnvVariables(containerStepGroupHelper.validateEnvVariables(envVarsWithSecretRef));
> 
>     PluginCreationResponse response =
> ```

The creation of PluginCreationResponse can be moved to a separate method and in a utility class.
___
*[`970-ng-commons/src/main/java/io/harness/entities/EntityType.java` line 703 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFx_hy1ge8ScWTQ1FL:-NlFx_hy1ge8ScWTQ1FM:bs2dqbo) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/970-ng-commons/src/main/java/io/harness/entities/EntityType.java#L703)):*
> ```Java
>       ModuleType.IDP, EntityTypeConstants.REGISTER_CATALOG, IdentifierRef.class, EntityYamlRootNames.REGISTER_CATALOG),
>   @JsonProperty(EntityTypeConstants.DOWNLOAD_HARNESS_STORE)
>   DOWNLOAD_HARNESS_STORE(ModuleType.CD, EntityTypeConstants.DOWNLOAD_HARNESS_STORE, IdentifierRef.class,
> ```

DOWNLOAD_HARNESS_STORE >> DOWNLOAD_HARNESS_FILE_STORE

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/containerStepGroup/DownloadHarnessStoreStepVariableCreator.java` line 23 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFu62s0Ge_mlBJYlXF:-NlM1KnG2ym4lGjVDXHE:b-zdmyp7) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/containerStepGroup/DownloadHarnessStoreStepVariableCreator.java#L23)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

The same comment for renaming DOWNLOAD_HARNESS_STORE >> DOWNLOAD_HARNESS_FILE_STORE
</blockquote></details>

Let's not change the type. Because in the tech spec, we have already decided for DownloadHarnessStore 

https://harness.atlassian.net/wiki/spaces/DEV/pages/21551744043/Download+Manifest+From+Harness+Store+in+CD+Container+Step+Group

It will cause a lot of changes. We'll change the class names but lets not change the types which is going to affect the spec

@ivan-83 Do let me know if its okay

**Comment by piyush-bhuwalka** [comment]
*[`970-ng-commons/src/main/java/io/harness/entities/EntityType.java` line 703 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFx_hy1ge8ScWTQ1FL:-NlM1yE60rnJppfy-FsN:b-zdmyp7) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/970-ng-commons/src/main/java/io/harness/entities/EntityType.java#L703)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

DOWNLOAD_HARNESS_STORE >> DOWNLOAD_HARNESS_FILE_STORE
</blockquote></details>

Let's not change the type. Because in the tech spec, we have already decided for DownloadHarnessStore 

https://harness.atlassian.net/wiki/spaces/DEV/pages/21551744043/Download+Manifest+From+Harness+Store+in+CD+Container+Step+Group

It will cause a lot of changes. We'll change the class names but lets not change the types which is going to affect the spec

@ivan-83 Do let me know if its okay

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1626 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFt5Rs2DpmGH05ZjfE:-NlM25d4-HMwSSdkWSy5:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1626)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

DOWNLOAD Harness Store >> Download Harness File Store"
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`120-ng-manager/config.yml` line 556 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFksuy4kuxxbLsI3Wx:-NlM29wR9yXqcsqsXn_Z:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/120-ng-manager/config.yml#L556)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Minor: download-harness-store >> download-harness-file-store or download-harness-fs
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFnO6o3hWMEFwN-whZ:-NlM2M3G0y5k29--usSu:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L38)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Minor: DownloadHarnessStoreStepHelper >> DownloadHarnessFileStoreStepHelper
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 62 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFmTJc0bLZFc4EHSMj:-NlM2WRt6BnT4m8ELIgK:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L62)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Minor: DownloadHarnessStoreStepParameters >> DownloadHarnessFileStoreStepParameters,
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java` line 45 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFnboRA1Ids2yn9rGt:-NlM2XqGCppZEbJ-xO29:b-zdmyp7) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/DownloadHarnessStoreStep.java#L45)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

I think it's important to emphasize what type of store (blob, file, queue) in our case it's a file. Pls, rename all paces accordingly.
</blockquote></details>

Let's not change the type. Because in the tech spec, we have already decided for DownloadHarnessStore 

https://harness.atlassian.net/wiki/spaces/DEV/pages/21551744043/Download+Manifest+From+Harness+Store+in+CD+Container+Step+Group

It will cause a lot of changes. We'll change the class names but lets not change the types which is going to affect the spec

@ivan-83 Do let me know if its okay

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by ivan-83 — CHANGES_REQUESTED** [review]
> Reviewed 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)* status: 34 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1626 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFt5Rs2DpmGH05ZjfE:-NlNAI7c-1dTNz-q5a_I:b3mkh60) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1626)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Done
</blockquote></details>

We haven't changed DOWNLOAD Harness Store >> Download Harness File Store.
___
*[`970-ng-commons/src/main/java/io/harness/entities/EntityType.java` line 703 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFx_hy1ge8ScWTQ1FL:-NlN9MdS4shEWS7B-r3g:bzbrw1s) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/970-ng-commons/src/main/java/io/harness/entities/EntityType.java#L703)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Let's not change the type. Because in the tech spec, we have already decided for DownloadHarnessStore 

https://harness.atlassian.net/wiki/spaces/DEV/pages/21551744043/Download+Manifest+From+Harness+Store+in+CD+Container+Step+Group

It will cause a lot of changes. We'll change the class names but lets not change the types which is going to affect the spec

@ivan-83 Do let me know if its okay
</blockquote></details>

Sure, just double-check with Rohan whether HarnessStore is enough because there are different types of stores, files, blobs....

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger ut5

**Comment by piyush-bhuwalka** [comment]
*[`970-ng-commons/src/main/java/io/harness/entities/EntityType.java` line 703 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFx_hy1ge8ScWTQ1FL:-NlNbFKMEz_Pi1OtYTyA:b6u2auo) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/970-ng-commons/src/main/java/io/harness/entities/EntityType.java#L703)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

Sure, just double-check with Rohan whether HarnessStore is enough because there are different types of stores, files, blobs....
</blockquote></details>

We have already reviewed the tech spec with Rohan . I believe we could add in the documentation about it. It allows us the flexibility to support more in future because we would use the existing step only in future for other harness store types

**Comment by piyush-bhuwalka** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1626 at r3](https://reviewable.io/reviews/harness/harness-core/56418#-NlFt5Rs2DpmGH05ZjfE:-NlNblDV4MT_aRfpYgYB:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/142fc75747229cda5ff384816120f97ff28e9e74/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1626)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

We haven't changed DOWNLOAD Harness Store >> Download Harness File Store.
</blockquote></details>

Done

**Review by ivan-83 — DISMISSED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)* status: 34 file(s) to review. 1 of 2 approvals obtained.

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java [hunk 1]
```diff
@@ -71,4 +79,25 @@ public Map<String, String> removeAllEnvVarsWithSecretRef(Map<String, String> env
...
+                                      .build();
+    return PluginCreationResponseWrapper.newBuilder().setResponse(response).setStepInfo(stepInfoProto).build();
+  }
+
+  public String convertToJson(Map<String, String> files) {
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(files);
+    } catch (JsonProcessingException e) {
+      throw new InvalidRequestException("Not able to convert list of files in Harness Store to json");
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java:100]: we should log the exception here ?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java [hunk 2]
```diff
@@ -1618,6 +1622,16 @@ public List<StepInfo> getStepInfo() {
...
                                  .build();
 
+    StepInfo downloadHarnessStore = StepInfo.newBuilder()
+                                        .setName("Download Harness File Store")
+                                        .setType(StepSpecTypeConstants.DOWNLOAD_HARNESS_STORE)
+                                        .setStepMetaData(StepMetaData.newBuilder()
+                                                             .addAllCategory(CD_STEP_GROUP_CONTAINER_STEPS_CATEGORY)
+                                                             .setFolderPath(CD_STEP_GROUP_CONTAINER_STEPS_METADATA)
+                                                             .build())
+                                        .setFeatureFlag(FeatureName.CDS_CONTAINER_STEP_GROUP_AWS_S3_DOWNLOAD.name())
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:1632]: We are using S3 FF for this ?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            ambiance, (DownloadHarnessStoreStepParameters) downloadHarnessStoreStepInfo.getSpecParameters(), null));
+
+    pluginDetailsBuilder.putAllEnvVariables(containerStepGroupHelper.validateEnvVariables(envVarsWithSecretRef));
+
+    return containerStepGroupHelper.getPluginCreationResponseWrapper(cdAbstractStepNode, pluginDetailsBuilder);
+  }
+
+  @Override
+  public boolean isSupported(String stepType) {
+    if (stepType.equals(StepSpecTypeConstants.DOWNLOAD_HARNESS_STORE)) {
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java:80]: nit: constants should be on the left side of comparison

we can re-write
```java
return StepSpecTypeConstants.DOWNLOAD_HARNESS_STORE,.equals(stepType)
```

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 4]
```diff
@@ -922,7 +922,7 @@ public enum FeatureName {
           + " We will fail the execution if we found some invalid configurations.",
       HarnessTeam.CDP),
   CDS_CONTAINER_STEP_GROUP_AWS_S3_DOWNLOAD(
-      "Enable to dowmload Aws S3 buckets, folder or object inside container step group", HarnessTeam.CDP),
+      "Enable to download Aws S3 buckets, folder or object inside container step group", HarnessTeam.CDP),
```
Comment (yogesh-chauhan) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:925]: should we update description to say that this is valid for Harness store also ? I think we need to update in UAT directly

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java:103]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java (hunk 2 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:1632]: Yes

File: 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java (hunk 3 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/DownloadHarnessStorePluginInfoProvider.java]: Done

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java (hunk 4 above)
Comment (piyush-bhuwalka) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:925]: I'll discuss this internally and if needed , I'll make the change in next PR for the desciption

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java` line 102 at r7](https://reviewable.io/reviews/harness/harness-core/56418#-NlQan6E-g-Eliz3I_4C:-NlQan6E-g-Eliz3I_4D:bs315nm) ([raw file](https://github.com/harness/harness-core/blob/7e7821dd68e5bc29e8c6540330c4bf57d54aa1e4/125-cd-nextgen/src/main/java/io/harness/cdng/containerStepGroup/ContainerStepGroupHelper.java#L102)):*
> ```Java
>       return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(files);
>     } catch (JsonProcessingException e) {
>       log.info("Not able to convert list of files in Harness Store to json :" + e.getMessage());
> ```

should be log.error
Also, its better to just do this
```
 throw new InvalidRequestException("Not able to convert list of files in Harness Store to json", ex);
```

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 4 of 33 files at r1, 2 of 9 files at r3, 6 of 6 files at r6, 1 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)* status: 28 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger ut5, ut6

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 1 files at r9.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418)* status: 28 file(s) to review. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by piyush-bhuwalka** [comment]
trigger ut5, ut7

**Comment by piyush-bhuwalka** [comment]
trigger ut5

**Comment by piyush-bhuwalka** [comment]
trigger ut7

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r6, 1 of 2 files at r7, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56418#-:-NlS9DCB5WKrbI_Gf16G:bnfp4nl)* status: 20 file(s) to review.

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by ankittiwari-harness — APPROVED** [review]

**Pull request merged**